File: CS/HideableGroupRowFooters/CustomGridControl/ShowFooterEventArgs.cs
using System;

namespace HideableGroupRowFooters
{
	public delegate void ShowGroupFooterEventHandler(object sender, ShowGroupFooterEventArgs e);

	public class ShowGroupFooterEventArgs : EventArgs
	{
		private int footerLevel;
        private bool footerVisible;
        private int handle;

		public ShowGroupFooterEventArgs(int aFooterLevel,int aHandle)
			: base()
		{
			footerVisible = true;
			footerLevel = aFooterLevel;
            handle = aHandle;
		}

		public int FooterLevel
		{
			get
			{
				return footerLevel;
			}
		}

        public bool Visible
        {
            get
            {
                return footerVisible;
            }
            set
            {
                footerVisible = value;
            }
        }
        public int Handle
        {
            get
            {
                return handle;
            }
        
        
        }
	}
}

File: VB/HideableGroupRowFooters/CustomGridControl/ShowFooterEventArgs.vb
Imports Microsoft.VisualBasic
Imports System

Namespace HideableGroupRowFooters
	Public Delegate Sub ShowGroupFooterEventHandler(ByVal sender As Object, ByVal e As ShowGroupFooterEventArgs)

	Public Class ShowGroupFooterEventArgs
		Inherits EventArgs
		Private footerLevel_Renamed As Integer
		Private footerVisible As Boolean
		Private handle_Renamed As Integer

		Public Sub New(ByVal aFooterLevel As Integer, ByVal aHandle As Integer)
			MyBase.New()
			footerVisible = True
			footerLevel_Renamed = aFooterLevel
			handle_Renamed = aHandle
		End Sub

		Public ReadOnly Property FooterLevel() As Integer
			Get
				Return footerLevel_Renamed
			End Get
		End Property

		Public Property Visible() As Boolean
			Get
				Return footerVisible
			End Get
			Set(ByVal value As Boolean)
				footerVisible = value
			End Set
		End Property
		Public ReadOnly Property Handle() As Integer
			Get
				Return handle_Renamed
			End Get


		End Property
	End Class
End Namespace
